pkg/apis/core/validation/validation.go

```diff
@@ -732,6 +732,14 @@ func validateVolumeSource(source *core.VolumeSource, fldPath *field.Path, volNam
 			}
 		}
 	}
+	if utilfeature.DefaultFeatureGate.Enabled(features.ImageVolume) && source.Image != nil {
```

Comment:
validation functions aren't supposed to directly depend on featuregates.  Instead, wire an options struct like is done for jobs here: https://github.com/kubernetes/kubernetes/blob/master/pkg/registry/batch/job/strategy.go#L346-L347 and https://github.com/kubernetes/kubernetes/blob/master/pkg/apis/batch/validation/validation.go#L152 .

Reply:
Yep, fixed as suggested.